src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java

```diff
@@ -213,10 +223,31 @@ private void onGetAnomalyDetectorResponse(GetResponse response) {
         try (XContentParser parser = RestHandlerUtils.createXContentParserFromRegistry(xContentRegistry, response.getSourceAsBytesRef())) {
             ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
             AnomalyDetector existingDetector = AnomalyDetector.parse(parser, response.getId(), response.getVersion());
-            if (!hasCategoryField(existingDetector) && hasCategoryField(this.anomalyDetector)) {
-                validateAgainstExistingMultiEntityAnomalyDetector(detectorId);
+            // We have separate flows for realtime and historical detector currently. User
+            // can't change detector from realtime to historical, vice versa.
+            if (!(existingDetector.isRealTimeDetector() ^ !anomalyDetector.isRealTimeDetector())) {
+                listener
+                    .onFailure(
+                        new ElasticsearchStatusException(
+                            "Can't change detector type between realtime and historical detector",
+                            RestStatus.BAD_REQUEST
+                        )
+                    );
+                return;
+            }
+
+            if (existingDetector.isRealTimeDetector()) {
+                validateDetector(existingDetector);
             } else {
-                validateCategoricalField(detectorId);
+                adTaskManager.getLatestADTask(detectorId, (adTask) -> {
+                    if (adTask.isPresent() && !adTaskManager.isADTaskEnded(adTask.get())) {
+                        // can't update detector if there is AD task running
+                        listener.onFailure(new ElasticsearchStatusException("Detector is running", RestStatus.INTERNAL_SERVER_ERROR));
+                    } else {
+                        // TODO: change to validateDetector method when we support HC historical detector
+                        searchAdInputIndices(detectorId);
+                    }
+                }, transportService, listener);
```

Comment:
I see the TODO for the change of validateDetector.  Can it be done in this PR?

Reply:
The `validateDetector` is only for HC detector. It will not validate single entity detector. Will change this method name to make it clear.